.gitlab-ci.yml

```diff
@@ -0,0 +1,22 @@
+image: fedora
+
+sync:
+
+  script:
+    - echo "Synchronizing $CI_COMMIT_BRANCH branch from $UPSTREAM_URL to $CI_PROJECT_URL"
+    - dnf install -y git
+    - git remote set-url origin https://sync:$ACCESS_TOKEN@$CI_SERVER_HOST/$CI_PROJECT_PATH.git
+    - git remote remove upstream || true
+    - git remote add upstream $UPSTREAM_URL || true
+    - git remote -v
+    - git fetch upstream $CI_COMMIT_BRANCH
+    - git checkout upstream/$CI_COMMIT_BRANCH
+    - git log -1
```

Comment:
If you push multiple commits upstream then this will only log the HEAD commit, so maybe instead: `git log upstream/$CI_COMMIT_BRANCH...origin/$CI_COMMIT_BRANCH --oneline` to get the full list of added commits?

Reply:
Good idea! I use `git log origin/$CI_COMMIT_BRANCH..upstream/$CI_COMMIT_BRANCH --oneline` since I'm assuming that `upstream` is ahead of `origin` so we only need the 2 dots. Is that OK?
https://stackoverflow.com/questions/462974/what-are-the-differences-between-double-dot-and-triple-dot-in-git-com

Reply:
Yeah makes sense :100: :